frame/support/src/lib.rs

```diff
@@ -1592,23 +1588,6 @@ pub mod pallet_prelude {
 /// only).
 /// For ease of use, bound the trait `Member` available in frame_support::pallet_prelude.
 ///
-/// Variant documentations and field types are put into metadata.
-/// The attribute `#[pallet::metadata(..)]` allows to specify the metadata to put for some types.
-///
```

Comment:
Are we sure this is fine to remove?
Doesn't this mean that pallets using have to use scale-info from this Substrate version on? Wouldn't that make a gradual transition impossible?
Would be cool to make a gradual migration possible. (Meaning that you can have scale-info and non-scale-info pallets in your runtime.)

Reply:
> Are we sure this is fine to remove?

Yes confirmed it witlh @thiolliere. The primary function of this attribute was to ensure identical metadata between FRAME v1 and v2 macros. However since this changes the metadata format entirely for all pallets, these are okay to remove.

> Doesn't this mean that pallets using have to use scale-info from this Substrate version on? Wouldn't that make a gradual transition impossible?

Not this specific change, but the whole metadata format for all pallets is changing.

> Would be cool to make a gradual migration possible. (Meaning that you can have scale-info and non-scale-info pallets in your runtime.)

I think that would be difficult and more trouble than it is worth, mixing old and new metadata.

Originally I did keep the legacy metadata stuff around, with the idea that you could ask for which version of the metadata you would like, but that also became unwieldy. It would also result in even more bloated wasm blobs.

If the legacy metadata is still required, then most likely the way to do it will be to provide a utility to convert `v14` to `v13` metadata.

See more discussion https://github.com/paritytech/substrate/discussions/8370